### PR TITLE
add Research Computing and CUmulus resources for University of Colorado - Boulder

### PR DESCRIPTION
**Pull request by bmshouse**
Please review my attempt to create the necessary files to add our resources to OpenScienceGrid, and let me know what needs to be fixed. Thank you.

### CODE REVIEW

File: topology/University of Colorado - Boulder/Research Computing/CUmulus.yaml [hunk 1]
```diff
@@ -0,0 +1,1366 @@
...
+# Otherwise, leave it blank and we will fill in the appropriate value for you.
+GroupID: <ID>
+
+# Resources contains one or more resources in this
+# ResourceGroup. A resource provides one or more services
+Resources:
+  # Resource Name should be a short descriptor of the resource.
+  # e.g. the Center for High Throughput Computing's GlideinWMS Frontend is "CHTC-glidein2"
+  # Resource Names need to be unique across all resources in the OSG.
+  porter1s1:
```
Comment (brianhlin) [inline topology/University of Colorado - Boulder/Research Computing/CUmulus.yaml:19]:
```suggestion
  CUmulus-Backfill:
```

File: topology/University of Colorado - Boulder/Research Computing/CUmulus.yaml [hunk 2]
```diff
@@ -0,0 +1,1366 @@
...
+    #   - <HOSTNAME2>
+
+    ### DN (optional except for XCache resources) is the DN of the host cert of the resource
+    # in OpenSSL 1.0 format (i.e. /DC=org/DC=incommon/C=US/...)
+    # DN: <DN>
+
+    # Services is one or more services provided by this resource;
+    # valid services are listed in topology/services.yaml with the format "<SERVICE NAME>: <ID>"
+    Services:
+      Virtual Machine Host:
```
Comment (brianhlin) [inline topology/University of Colorado - Boulder/Research Computing/CUmulus.yaml:71]:
```suggestion
      CE:
```

We don't yet have a dedicated service type for the backfill containers so we're going to use "CE" to make sure that the accounting shows up properly

File: topology/University of Colorado - Boulder/Research Computing/CUmulus.yaml [hunk 3]
```diff
@@ -0,0 +1,1366 @@
...
+        #   # endpoint: <HOSTNAME>
+
+      # Other services if you have any
+      # <SERVICE NAME>:
+      # ...
+
+    ### Tags (optional) is a list of tags associated with the resource.
+    ### Include the tag "CC*" if applicable for a CC* CE.
+    # Tags:
+    #   - <TAG1>
```
Comment (brianhlin) [inline topology/University of Colorado - Boulder/Research Computing/CUmulus.yaml:92]:
```suggestion
    Tags:
      - CC*
```

Tagging this as a CC* resource

File: topology/University of Colorado - Boulder/Research Computing/CUmulus.yaml [hunk 4]
```diff
@@ -0,0 +1,1366 @@
...
+    #   InteropMonitoring: true
+    #   KSI2KMax: 0
+    #   KSI2KMin: 0
+    #   StorageCapacityMax: 0
+    #   StorageCapacityMin: 0
+    #   TapeCapacity: 0
+
+  # Other resources if you have any...
+  # <RESOURCE NAME>:
+  # ...
```
Comment (brianhlin) [inline topology/University of Colorado - Boulder/Research Computing/CUmulus.yaml:1366]:
```suggestion
```

Confusingly, we don't need to register each of the VMs in question. Just a single "endpoint" is fine for our purposes!

File: topology/University of Colorado - Boulder/Research Computing/CUmulus.yaml [hunk 5]
```diff
@@ -0,0 +1,1366 @@
...
+    ### DN (optional except for XCache resources) is the DN of the host cert of the resource
+    # in OpenSSL 1.0 format (i.e. /DC=org/DC=incommon/C=US/...)
+    # DN: <DN>
+
+    # Services is one or more services provided by this resource;
+    # valid services are listed in topology/services.yaml with the format "<SERVICE NAME>: <ID>"
+    Services:
+      Virtual Machine Host:
+        # Description is a brief description of the service
+        Description: a compute node hosting virtual machines
```
Comment (brianhlin) [inline topology/University of Colorado - Boulder/Research Computing/CUmulus.yaml:73]:
```suggestion
        Description: Backfill containers running on CUmulus nodes
```

**Review by brianhlin — COMMENTED** [review]
Hi @bmshouse, thanks for the PR! I've made a few suggestions that simplify this registration: most notably, we don't need to register each of the nodes for which you plan to run a backfill container. We can group them all under a single "resource".

I have to run a question about the new facility up the chain (there's an existing `University of Colorado` facility that appears to physically exist at Boulder) but once I get an answer for that, we can move forward with this registration.

**Comment by bmshouse** [comment]
Thank you for taking a look at this a getting things straightened out! I am fine with using the existing University of Colorado. Boulder is the main campus with 3 other locations { https://www.cu.edu/ }.

**Pull request merged**